OSDOCS-19969: Gateway API backendTLSPolicy for securing HTTPRoutes (#112677)
openshift/openshift-docs: modules/creating-httproute.adoc

// Module included in the following assemblies:
//
// * networking/ingress_load_balancing/configuring_ingress_cluster_traffic/routing-http-requests-to-services.adoc

:_mod-docs-content-type: PROCEDURE
[id="creating-httproute_{context}"]
= Creating a basic HTTPRoute custom resource

[role="_abstract"]
To direct incoming network traffic from a gateway to your backend applications, you must create an `HTTPRoute` custom resource (CR). The resource specifies the hostnames the route handles and binds them to a parent gateway.

.Prerequisites
* You have a target backend service running on your cluster.

> * You know the name and namespace of the deployed `Gateway` resource.

Reviewer: ```suggestion
* You know the name and namespace of the deployed `Gateway` that serves as the parent of the HTTPRoute.
```

Author: Got it, thanks!

.Procedure

> * Create an `HTTPRoute` CR file that defines your application hostnames and points to your parent gateway, and save it as `httproute.yaml`:

Reviewer: ```suggestion
* Create an `HTTPRoute` CR file that references your parent gateway (parentRef), lists your application hostnames, and includes a backendRef rule that points to your backend service, and save it as `httproute.yaml`:
```

Reviewer: Also, referring to a previous [comment](https://github.com/openshift/openshift-docs/pull/112677/changes#r3431523333), if the HTTPRoute has to be in the same namespace as the Gateway, then this is incorrect.  Has this yaml been tested?

Reviewer 2: this is fine, as long as the gateway listener is configured like so
```
  listeners:
  - name: demo
    hostname: "*.example.com"
    port: 80
    protocol: HTTP
    allowedRoutes: <--
      namespaces: <--
        from: All <--
```
without these sections the gateway will reject the `HTTPRoute`

Author: Excellent! I've updated the prerequisite and the lead-in sentence using your suggested phrasing. I just added the backticks around literal API fields.

Also, excellent point regarding the namespace discrepancy in the YAML. I've added a Note block above the YAML to warn users about this gateway requirement.

// Module included in the following assemblies:
//
// * networking/ingress_load_balancing/configuring_gateway_api/routing-http-requests-to-services.adoc

:_mod-docs-content-type: PROCEDURE
[id="creating-httproute_{context}"]
= Creating a basic HTTPRoute custom resource

[role="_abstract"]
To direct incoming network traffic from a gateway to your backend applications, you must create an `HTTPRoute` custom resource (CR). The resource specifies the hostnames the route handles and binds them to a parent gateway.

.Prerequisites

> * You have a target backend service running on your cluster.

Reviewer 3: ```suggestion

* You have a target backend service running on your cluster.
```

> * You know the name and namespace of the deployed `Gateway` resource that serves as the parent of the `HTTPRoute`.

Reviewer 3: ```suggestion
* You know the name and namespace of the deployed `Gateway` custom resource (CR) that serves as the parent of the `HTTPRoute`.
```

.Procedure

. Create an `HTTPRoute` CR file that references your parent gateway (`parentRef`), lists your application hostnames, and includes a `backendRef` rule that points to your backend service, and save it as `httproute.yaml`:
+

> [NOTE]

Reviewer 3: ```suggestion
```

> ====

Reviewer 3: ```suggestion
```

> If your `HTTPRoute` and `Gateway` resources are deployed in different namespaces, the gateway listener must be configured to allow cross-namespace routes. For example, the gateway administrator must set `allowedRoutes.namespaces.from: All`. Otherwise, the gateway will reject the route.

Reviewer 3: ```suggestion
```

> ====

Reviewer 3: ```suggestion
```

+
[source,yaml]
----
apiVersion: gateway.networking.k8s.io/v1
kind: HTTPRoute
metadata:
  name: sample-route
  namespace: my-application
spec:
  parentRefs:
  - name: generic-gateway
    namespace: openshift-ingress
  hostnames:
  - app1.example.com
  - app2.example.com
  rules:
  - backendRefs:
    - name: application-backend
      port: 8080
----

> 

Reviewer 3: ```suggestion
+
* You must include application hostnames and a `backendRef` rule that points to your backend service.
* If your `HTTPRoute` and `Gateway` CRs are deployed in different namespaces, the `Gateway` CR listener must be configured to allow cross-namespace routes. You must set `allowedRoutes.namespaces.from: All` in the `Gateway` CR. 
```

Author: Love this suggestion, thanks!

. Apply the `HTTPRoute` CR file to your cluster:
+
[source,terminal]
----
$ oc apply -f httproute.yaml
----

.Verification

> * Verify that the `HTTPRoute` resource was successfully created and bound to the gateway:

Reviewer 3: ```suggestion
* Verify that the `HTTPRoute` resource was successfully created and bound to the `Gateway` CR by running the following command:
```

+
[source,terminal]
----

> $ oc get httproute sample-route -n my-application

Reviewer 3: ```suggestion
$ oc get httproute <sample_route> -n <my_application>
```
